src/robot/running/context.py

```diff
@@ -126,14 +126,22 @@ def in_teardown(self):
     def variables(self):
         return self.namespace.variables
 
-    @property
-    def continue_on_failure(self):
-        parents = ([self.test] if self.test else []) + self.user_keywords
+    def continue_on_failure(self, default=False):
+        if self.in_teardown:
+            parents = self.user_keywords
+        else:
+            parents = ([self.test] if self.test else []) + self.user_keywords
```

Comment:
Is the reason to handle teardowns differently that then parent test possibly having `robot:(recursive-)stop-on-failure` doesn't affect teardowns? I guess that makes sense, but then again we could consider making it possible to alter the default teardown behavior with that tag a feature. Because stopping is the default behavior (outside teardowns), I see no reason to enable the stop-on-failure on test level otherwise.

Reply:
Here I followed your [guidance](https://github.com/robotframework/robotframework/issues/4303#issuecomment-1096889806):
> - Neither stop-on-failure variants should propagate to teardowns. We can think them having robot:recursive-continue-on-failure implicitly. To enable the stop-on-failure mode in teardowns, these special tags need to be used explicitly.
but we can reconsider?